lib/DxcSupport/dxcapi.extval.cpp

```diff
@@ -6,37 +6,192 @@
 
 namespace dxc {
 
+HRESULT STDMETHODCALLTYPE ExternalValidationHelper::Compile(
+    _In_ IDxcBlob *pSource,         // Source text to compile.
+    _In_opt_z_ LPCWSTR pSourceName, // Optional file name for pSource. Used
+    // in errors and include handlers.
+    _In_opt_z_ LPCWSTR pEntryPoint, // Entry point name.
+    _In_z_ LPCWSTR pTargetProfile,  // Shader profile to compile.
+    _In_opt_count_(argCount)
+        LPCWSTR *pArguments, // Array of pointers to arguments.
+    _In_ UINT32 argCount,    // Number of arguments.
+    _In_count_(defineCount) const DxcDefine *pDefines, // Array of defines.
+    _In_ UINT32 defineCount,                           // Number of defines.
+    _In_opt_ IDxcIncludeHandler
+        *pIncludeHandler, // User-provided interface to handle #include
+    // directives (optional).
+    _COM_Outptr_ IDxcOperationResult *
+        *ppResult // Compiler output status, buffer, and errors.
+) {
+  // First, add -Vd to the compilation args
+  UINT32 newArgCount = argCount + 1;
+  std::vector<LPCWSTR> newArgs;
+  newArgs.reserve(newArgCount);
+
+  // Copy existing args
+  for (unsigned int i = 0; i < argCount; ++i) {
+    newArgs.push_back(pArguments[i]);
+  }
+
+  // Add an extra argument
+  newArgs.push_back(L"-Vd");
+
+  return m_pCompiler->Compile(pSource, pSourceName, pEntryPoint, pTargetProfile,
+                              newArgs.data(), newArgCount, pDefines,
+                              defineCount, pIncludeHandler, ppResult);
+}
+
+// Implement the IDxcCompiler3 interface Compile command
+HRESULT STDMETHODCALLTYPE ExternalValidationHelper3::Compile(
+    _In_ const DxcBuffer *pSource, ///< Source text to compile.
+    _In_opt_count_(argCount)
+        LPCWSTR *pArguments, ///< Array of pointers to arguments.
+    _In_ UINT32 argCount,    ///< Number of arguments.
+    _In_opt_ IDxcIncludeHandler
+        *pIncludeHandler, ///< user-provided interface to handle include
+    ///< directives (optional).
+    _In_ REFIID riid,      ///< Interface ID for the result.
+    _Out_ LPVOID *ppResult ///< IDxcResult: status, buffer, and errors.
+) {
+  // First, add -Vd to the compilation args
+  UINT32 newArgCount = argCount + 1;
+  std::vector<LPCWSTR> newArgs;
+  newArgs.reserve(newArgCount);
+
+  // Copy existing args
+  for (unsigned int i = 0; i < argCount; ++i) {
+    newArgs.push_back(pArguments[i]);
+  }
+
+  // Add an extra argument
+  newArgs.push_back(L"-Vd");
```

Comment:
Seems ripe to pull out into a `std::vector<LPCWSTR> EnsureDisableValidation(....)` style helper.